pkg/tfvars/internal/cache/cache.go

```diff
@@ -21,6 +23,30 @@ const (
 	imageDataType   = "image"
 )
 
+type reader func(io.Reader) (*io.Reader, error)
+type closer func(io.Reader) error
+
+type decompressor struct {
+	reader reader
+	closer closer
+}
+
+func nullCloser(_ io.Reader) error {
+	return nil
+}
+
+var decompressorMap = map[string]decompressor {
+	"application/x-gzip": {
+		reader: newGzipReader, 
+		closer: nullCloser,
+	},
+}
+
+func newGzipReader(reader io.Reader) (*io.Reader, error) {
+	reader, err := gzip.NewReader(reader)
+	return &reader, err
+}
```

Comment:
if we drop the `getFileFromCache` changes as they shouldn't be required..

in which case i would like to maybe move to a switch at the call site.. it would be much simpler IMO.
gzip has a Closer function at it shouldn't be `nullCloser`

Reply:
The idea here was to allow adding other decompressors later.

Reply:
Going ahead and making this change; it's all in one place now anyway.